File: JAVA Codes/archive/unsorted/2017.02/2017.02.12 - USACOFEB2017/helpcross.java
package juniorAndrade.main;

import juniorAndrade.lib.collections.Pair;
import juniorAndrade.lib.collections.set.MultiSet;
import juniorAndrade.lib.utils.io.InputReader;
import juniorAndrade.lib.utils.io.OutputWriter;

import java.util.Arrays;
import java.util.Iterator;

public class helpcross {
    public void solve(int testNumber, InputReader in, OutputWriter out) {
        int n = in.readInt();
        int m = in.readInt();
        MultiSet<Integer> s = new MultiSet<Integer>();
        for(int i = 0; i < n; ++i) {
            int x = in.readInt();
            s.add(x);
        }
        Pair<Integer, Integer>[] arr = new Pair[m];
        for(int i = 0; i < m; ++i) {
            int x = in.readInt();
            int y = in.readInt();
            arr[i] = Pair.makePair(y, x);
        }
        Arrays.sort(arr);
        int ans = 0;
        for(int i = 0; i < m; ++i) {
            int l = arr[i].second;
            int r = arr[i].first;
            Iterator<Integer> it = s.low
        }
        out.printLine(ans);
    }
}

File: USACO/helpcross.cpp
#include <bits/stdc++.h>

using namespace std;

typedef long long ll;
typedef long double ld;
typedef unsigned long long ull;
typedef pair< ll, ll > ii;
typedef vector< ll > vi;
typedef vector< ii > vii;

#define INF 0x3F3F3F3F
#define LINF 0x3F3F3F3F3F3F3F3FLL
#define pb push_back
#define mp make_pair
#define pq priority_queue
#define LSONE(s) ((s)&(-s))
#define DEG_to_RAD(X)   (X * PI / 180)
#define F first
#define S second

#ifdef ONLINE_JUDGE
#define debug(args...)
#else
#define debug(args...) fprintf(stderr,args)
#endif

void arquivo() {
  freopen("helpcross.in", "r", stdin);
  freopen("helpcross.out", "w", stdout);
}

const int N = 20010;

int n, m;
multiset<int>s;
ii arr[N];

int main() {
  arquivo();
  scanf("%d %d", &n, &m);
  for(int i = 0; i < n; ++i) {
    int x; scanf("%d", &x);
    s.insert(x);
  }
  for(int i = 0; i < m; ++i) {
    int x, y; scanf("%d %d", &x, &y);
    arr[i] = mp(y, x);
  }
  sort(arr, arr + m);
  int ans = 0;
  for(int i = 0; i < m; ++i) {
    int l, r;
    l = arr[i].S;
    r = arr[i].F;
    multiset<int>::iterator it = s.lower_bound(l);
    if(it == s.end()) continue;
    if(*it >= l && *it <= r) {
      ans++;
      s.erase(it);
    }
  }
  printf("%d\n", ans);
  return 0;
}